File: backend/app/schemas/order.py
"""注文書スキーマ。"""
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.models.enums import OrderStatus


class OrderCreate(BaseModel):
    issue_date: date | None = None
    client_address: str | None = None
    client_company: str | None = None
    client_person: str | None = None
    amount_excl_tax: float | None = None
    tax_amount: float | None = None
    total_amount: float | None = None
    construction_period_start: date | None = None
    construction_period_end: date | None = None
    payment_condition: str | None = None
    terms_and_conditions: str | None = None


class OrderUpdate(OrderCreate):
    status: OrderStatus | None = None


class OrderRead(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    order_number: str | None
    issue_date: date | None
    client_address: str | None
    client_company: str | None
    client_person: str | None
    amount_excl_tax: float | None
    tax_amount: float | None
    total_amount: float | None
    construction_period_start: date | None
    construction_period_end: date | None
    payment_condition: str | None
    terms_and_conditions: str | None
    stamp_tax: float | None
    quote_id: uuid.UUID | None
    linked_to_quote: bool
    status: OrderStatus
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/order.ts
export type OrderStatus = "draft" | "sent" | "signed" | "acknowledged" | "cancelled";

export const ORDER_STATUS_LABEL: Record<OrderStatus, string> = {
  draft: "下書き",
  sent: "発行済み",
  signed: "サイン受領済",
  acknowledged: "注文請書発行済",
  cancelled: "キャンセル",
};

export interface OrderCreate {
  issue_date?: string | null;
  client_address?: string | null;
  client_company?: string | null;
  client_person?: string | null;
  amount_excl_tax?: number | null;
  construction_period_start?: string | null;
  construction_period_end?: string | null;
  payment_condition?: string | null;
  work_content?: string | null;
  notes?: string | null;
  terms_and_conditions?: string | null;
}

export interface OrderRead {
  id: string;
  project_id: string;
  order_number: string | null;
  issue_date: string | null;
  client_address: string | null;
  client_company: string | null;
  client_person: string | null;
  amount_excl_tax: number | null;
  tax_amount: number | null;
  total_amount: number | null;
  construction_period_start: string | null;
  construction_period_end: string | null;
  payment_condition: string | null;
  work_content: string | null;
  notes: string | null;
  terms_and_conditions: string | null;
  stamp_tax: number | null;
  quote_id: string | null;
  linked_to_quote: boolean;
  status: OrderStatus;
  created_at: string;
  updated_at: string;
}
